asreview/webapp/auth.py

```diff
@@ -0,0 +1,144 @@
+    user = User.query.filter(User.username == username).one_or_none()
+    # if the user exist proceed and verify
+    if not user:
+        result = (404, {'message': f'User account {username} does not exist.'})
+    else:
+        # verify password
+        if user.verify_password(password):
+            logged_in = login_user(
+                user,
+                remember=True,
+                duration=datetime.timedelta(days=31)
+            )
+            result = (200, {
+                'logged_in': logged_in,
+                'username': username,
+                'id': user.id
+            })
+        else:
+            # password is wrong
+            result = (
+                404,
+                {'message': f'Incorrect password for user {username}'}
+            )
+
+    status, message = result
+    # check in database if username exists and if password is correct
+    response = jsonify(message)
+    return response, status
+
+
+@bp.route('/signup', methods=["POST"])
+def signup():
+    print('-------->')
+    print(request.form)
+    username = request.form.get('username').strip().lower()
+    password = request.form.get('password')
+    first_name = request.form.get('first_name', '').strip()
+    last_name = request.form.get('last_name', '').strip()
+    affiliation = request.form.get('affiliation', '').strip()
+    email = request.form.get('email', '').strip()
+    public = bool(int(request.form.get('public', '0')))
+
+    # check if username already exists
+    user = User.query.filter(User.username == username).one_or_none()
+    # return error if user doesn't exist
+    if isinstance(user, User):
+        result = (404, f'Username "{username}" already exists.')
+    else:
+        # password confirmation is done by front end, so the only
+        # thing that remains is to add the user (password will be
+        # hashed in the User model)
+        try:
+            user = User(username, password, first_name, last_name,
+                affiliation, email, public)
+            DB.session.add(user)
+            DB.session.commit()
+            # at this stage we know the user account is stored,
+            # let's add a working directory for this user
+            Path(asreview_path(), str(user.id)).mkdir(exist_ok=True)
+            # result is a 201 with message
+            result = (201, f'User "#{username}" created.')
+        except SQLAlchemyError:
+            DB.session.rollback()
+            result = (500, 'Creating account unsuccessful!')
+
+    (status, message) = result
+    response = jsonify({'message': message, 'user_id': user.id})
+    return response, status
+
+
+@bp.route('/refresh', methods=["GET"])
+@asreview_login_required
+def refresh():
+    if current_user:
+        result = (200, {
+            'logged_in': current_user.is_authenticated,
+            'username': current_user.username,
+            'id': current_user.id
+        })
+    else:
+        result = (404, 'No user found')
```

Comment:
If `asreview_login_required` is used, can we skip this? Also, `asreview_login_required` returns 401 is there is no user, this returns 404.